Here is where I am at with Module 4a

tylerw8 opened the pull request:
Here is where I am at with Module 4a.  I need to go through and comment out the stuff that is for 4b, and finish two of the functions.

tylerw8 committed:
My progress so far with Module 4a

My progress on Module 4a as of May 3, 2015.

tylerw8 committed:
Update to app js

tylerw8 committed:
Removed Node Modules

.gitignore not working

jamdon commented on app.js:
```diff
@@ -1,4 +1,58 @@
 //  in this file create an express application - use the middle-ware built into express
+var express = require('express');
+var trucks = require('./trucks');
+var idManager = require('./idManager');
```
While you certainly can use this if you want to, it's not required. I used it in the in-class example to preview functionality that will ultimately be provided by MongoDB for us. If you don't want to mess with this, you can just create new food trucks without assigning them an Id. As it stands, implementing the POST route isn't required for the assignment for Module_4a.

jamdon commented on app.js:
```diff
@@ -1,4 +1,58 @@
+app.get('/trucks/:id', function (request, response) {
+        var id = request.params.id;
+  
+        var truck = trucks.getTruck(id);
```
Since the existing truck data we currently have doesn't have any id's , this won't return anything except for any new trucks you create and add to the foodTrucks array. Instead, for both this assignment and the Module_4b assignment, use "name" as the dynamic route parameter: `app.get('/trucks/:name', function (request, response) { var truckName = request.params.name; var truck = trucks.getTruck(truckName); if (truck) { response.json(truck); } else { response.status(404).json('Truck not found'); } });

**EDIT**: I just now realized you added id's, so if you want to do it this way instead, that's fine too.

jamdon commented on app.js:
```diff
@@ -1,4 +1,58 @@
+        }
+});
+
+app.post('/Trucks', function (request, response){
```
Assignment for Module_4a does not need the POST route implemented (that's Module_4b.)

jamdon commented on app.js:
```diff
@@ -1,4 +1,58 @@
+  }
+});
+
+app.delete('/trucks/:id', function (request, response) {
```
Assignment for Module_4b does not need the DELETE route implemented (that's for Module_4b.)

jamdon commented:
Let me know if you want this graded as is or if you're going to be making more changes as your initial pull request comment indicates.

tylerw8 commented:
I'll work on fixing today and starting on Module 5.  Thanks!

Sent from my iPhone

> On May 9, 2015, at 11:40 PM, jamdon notifications@github.com wrote:
> 
> Let me know if you want this graded as is or if you're going to be making more changes as your initial pull request comment indicates.
> 
> —
> Reply to this email directly or view it on GitHub.

tylerw8 committed:
Update of Module_4a based on in class demo

Updated Module_4a based on the example shown in class.  Module was
tested and performed as demonstrated in class.

tylerw8 committed:
Removed unneeded idManager and package.json

removed files not need for pull request

tylerw8 commented:
Ok, I fixed the code based on your in class demonstration, tested and uploaded it.  Should I close this pull and do a new pull?

jamdon commented:
No need for additional pull requests. Once you update the repository you initiated the original pull request from, I automatically see the changed files. You didn't want to remove the package.json file, just the parts of the application that were for the following week, like the body-parser module. Don't worry about it - everything is fine as is. :+1:
